include/thread.h

```diff
@@ -28,6 +28,7 @@ typedef struct thread {
   enum { TDS_INACTIVE = 0x0, TDS_WAITING, TDS_READY, TDS_RUNNING } td_state;
   uint32_t td_flags;           /* TDF_* flags */
   volatile uint32_t td_csnest; /* critical section nest level */
+  int td_exitcode;
```

Comment:
I suppose that would be okay, although the POSIX standard requires `waitid()` to recover the full number passed to `void exit(int status)`, which would mean we need all 32 bits for userspace exit codes.